pkg/server/handler/v1alpha1/resourcetypes.go

```diff
@@ -11,49 +11,21 @@ import (
 
 	"github.com/caicloud/cyclone/pkg/apis/cyclone/v1alpha1"
 	"github.com/caicloud/cyclone/pkg/common"
-	"github.com/caicloud/cyclone/pkg/meta"
 	svrcommon "github.com/caicloud/cyclone/pkg/server/common"
 	"github.com/caicloud/cyclone/pkg/server/handler"
 	"github.com/caicloud/cyclone/pkg/server/types"
 	"github.com/caicloud/cyclone/pkg/util/cerr"
+	wfcommon "github.com/caicloud/cyclone/pkg/workflow/common"
 )
 
-func getResourceTypes(namespaces []string, operation string) ([]v1alpha1.Resource, error) {
-	var results []v1alpha1.Resource
-	for _, ns := range namespaces {
-		resources, err := handler.K8sClient.CycloneV1alpha1().Resources(ns).List(metav1.ListOptions{
-			LabelSelector: meta.ResourceTypeSelector(),
-		})
-		if err != nil {
-			log.Errorf("Get resource type from namespace %s error: %v", ns, err)
-			return nil, cerr.ConvertK8sError(err)
-		}
-		for _, item := range resources.Items {
-			if operation == "" {
-				results = append(results, item)
-				continue
-			}
-
-			for _, op := range item.Spec.SupportedOperations {
-				if strings.ToLower(operation) == strings.ToLower(op) {
-					results = append(results, item)
-					break
-				}
-			}
-		}
-	}
-
-	return results, nil
-}
-
 // ListResourceTypes ...
 func ListResourceTypes(ctx context.Context, tenant string, operation string) (*types.ListResponse, error) {
 	namespaces := []string{svrcommon.TenantNamespace(tenant)}
 	if tenant != svrcommon.DefaultTenant {
 		namespaces = append(namespaces, common.GetSystemNamespace())
 	}
 
-	resources, err := getResourceTypes(namespaces, operation)
+	resources, err := wfcommon.GetResourceTypes(handler.K8sClient, namespaces, operation)
```

Comment:
It seems that there is only one `handler.K8sClient`, why not put this into `GetResourceTypes()`, then there is no need to pass anywhere.

At the same time, I think `GetResourceTypes()` is better to kept in this file, rather than moved into workflow common package.

Reply:
`common` provides utility, shouldn't depends on global variables like `handler.K8sClient`, otherwise it can't be used in other places if `handler.K8sClient` not initialized.

> I think GetResourceTypes() is better to kept in this file, rather than moved into workflow common package.

It will also be used in workflow engine, workflow engine shouldn't call top layer functions.

Reply:
> `common` provides utility, shouldn't depends on global variables like `handler.K8sClient`, otherwise it can't be used in other places if `handler.K8sClient` not initialized.

Make sense.

> It will also be used in workflow engine, workflow engine shouldn't call top layer functions.

It is used by server and workflow engine, Why not put inot `pkg/common`?

Reply:
Resource Type is concept of Workflow Engine, it's natural to be defined in Workflow Engine layer. And Cyclone Server already use lots pkg from Workflow Engine. I'd like to make top level 'pkg/common' lightweight

Reply:
ok, I think we need to uniformly handle these common packages later.